lib/directives/tooltip.js

```diff
@@ -0,0 +1,153 @@
+    }
+
+    // If Argument, assume element ID of container element
+    if (bindings.arg) {
+        // Element ID specified as arg. We must prepend '#' to become a CSS selector
+        config.container = `#${bindings.arg}`;
+    }
+
+    // Process modifiers
+    keys(bindings.modifiers).forEach(mod => {
+        if (/^html$/.test(mod)) {
+            // Title allows HTML
+            config.html = true;
+        } else if (/^nofade$/.test(mod)) {
+            // no animation
+            config.animation = false;
+        } else if (/^(auto|top|bottom|left|right)$/.test(mod)) {
+            // placement of tooltip
+            config.position = mod;
+        } else if (/^d\d+$/.test(mod)) {
+            // delay value
+            const delay = parseInt(mod.slice(1), 10) || 0;
+            if (delay) {
+                config.delay = delay;
+            }
+        } else if (/^o\d+$/.test(mod)) {
+            // offset value
+            const offset = parseInt(mod.slice(1), 10) || 0;
+            if (offset) {
+                config.offset = offset;
+            }
+        }
+    });
+
+    // Special handling of event trigger modifiers Trigger is a space separated list
+    const selectedTriggers = {};
+
+    // parse current config object trigger
+    let triggers = typeof config.trigger === 'string' ? config.trigger.trim().split(/\s+/) : [];
+    triggers.forEach(trigger => {
+        if (validTriggers[trigger]) {
+            selectedTriggers[trigger] = true;
+        }
+    });
+
+    // Parse Modifiers for triggers
+    keys(validTriggers).forEach(trigger => {
+        if (bindings.modifiers[trigger]) {
+            selectedTriggers[trigger] = true;
+        }
+    });
+
+    // Sanitize triggers
+    config.trigger = keys(selectedTriggers).join(' ');
+    if (!config.trigger) {
+        // remove trigger config
+        delete config.trigger;
+    }
+
+    return config;
+}
+
+//
+// Add or Update tooltip on our element
+//
+function applyBVTT(el, bindings, vnode) {
+    if (!inBrowser) {
+        return;
+    }
+    if (!Popper) {
+        // Popper is required for tooltips to work
+        warn("v-b-tooltip: Popper.js is required for tooltips to work");
+        return;
+    }
+    if (el[BVTT]) {
+        el[BVTT].updateConfig(parseBindings(bindings));
+    } else {
+        el[BVTT] = new ToolTip(el, parseBindings(bindings), vnode.context.$root);
+    }
+};
+
+//
+// Remove tooltip on our element
+//
+function removeBVTT(el) {
+    if (!inBrowser) {
+        return;
+    }
+    if (el[BVTT]) {
+        el[BVTT].destroy();
+        el[BVTT] = null;
+        delete el[BVTT];
+    }
+}
+
+/*
+ * Export our directive
+ */
+export default {
+    bind(el, bindings, vnode) {
+        applyBVTT(el, bindings, vnode);
+    },
+    inserted(el, bindings, vnode) {
+        applyBVTT(el, bindings, vnode);
+    },
+    update(el, bindings, vnode) {
+        if (bindings.oldValue !== bindsings.oldValue) {
```

Comment:
??? maybe 
`if (bindings.value !== bindings.oldValue) {`

Reply:
ooops... Yep.  So used to watchers